File: benches/ordered_set.cpp
#include <set>
#include <iostream>
#include <cstdint>
#include <ranges>

struct MyRand {
    uint64_t seed = 5;

    uint64_t next() {
        seed = seed * 123456789 + 101112131415;
        return seed;
    }
};

using namespace std;

int main() {
    MyRand rand{};
    set<uint64_t> hashset;
    for (int _: views::iota(0, 1'000'000)) {
        hashset.insert(rand.next());
    }
    cout << hashset.size() << endl;
}

File: benches/ordered_set.rs
use std::collections::BTreeSet;

#[derive(Default)]
struct MyRand {
    seed: u64,
}

impl MyRand {
    fn next(&mut self) -> u64 {
        self.seed = self.seed.overflowing_mul(123456789).0.overflowing_add(101112131415).0;
        self.seed
    }
}


fn main() {
    let mut rand = MyRand::default();
    let mut hashset = BTreeSet::new();
    for _ in 0..1_000_000 {
        hashset.insert(rand.next());
    }
    println!("{}", hashset.len());
}